### ISSUE TITLE
Bug Report: unable to complete guide for Vitess Operator for Kubernetes with current resources recommendations

### ISSUE THREAD

**Issue by vinimdocarmo** [vitessio/vitess#13118]
### Overview of the Issue

I’m following this guide ([Vitess Operator for Kubernetes](https://vitess.io/docs/16.0/get-started/operator/)) and I’m able to create the operator (`kubectl apply -f operator.yaml`) and the initial cluster (`kubectl apply -f 101_initial_cluster.yaml`), but when I check the pods all vttablet containers are not ready due to no connection to MySQL. I checked MySQL container logs and I could see the following error over and over again:

```
Mysqld.Start([1683641048](tel:1683641048)) stderr: Killed
```

After [some discussion in Slack](https://vitess.slack.com/archives/C02JQQ88Z1C/p1683631404400479), we found out that was a memory issue, which led us to believe that MySQL server - which is now version 8.0 by default - didn't have enough memory to start correctly.

### Local Resolution

If I change [this value](https://github.com/vitessio/vitess/blob/main/examples/operator/101_initial_cluster.yaml#L106) to `1024Mi` then all the containers/pods became ready and the tablets in a serving state. I see that planetscale/vitess-operator uses `1024Mi` in [their example](https://github.com/planetscale/vitess-operator/blob/main/test/endtoend/operator/101_initial_cluster_vtorc_vtadmin.yaml#L98) already.

From there I was able to continue the aforementioned guide ’til [this section](https://vitess.io/docs/16.0/user-guides/configuration-advanced/resharding/#using-operator-1) when running this command: `kubectl apply -f 302_new_shards.yaml`. With the caveat that now all my `mysqld` containers have `1024Mi` as memory limit. When I run that command, the initial 8GB for my minikube cluster seems to not be enough anymore and it starts getting really slow and freezing most of the time.

I deleted the cluster and started it again with a higher memory: `minikube start --kubernetes-version=v1.24.0 --cpus=4 --memory=12g --disk-size=32g`. With this new memory for my cluster and having `mysqld` with `1024Mi` memory limit I was able to complete the reshard section. When I checked minikube memory usage by the end it was floating around 9.5GB.

I am happy to create PRs to change those docs if that turns out to be needed.

### Reproduction Steps

Try to follow this [guide](https://vitess.io/docs/17.0/get-started/operator/) and after you run `kubectl apply -f 101_initial_cluster.yaml` you should be able to see that your vttablets are not serving and each of them will have 2/3 ready containers.

### Binary Version

```sh
$ vtgate --version
Version: 17.0.0-SNAPSHOT (Git revision 74548648c5f27234acfa27fa8a26f588d21af405 branch 'main') built on Thu May 18 20:35:12 UTC 2023 by vitess@buildkitsandbox using go1.20.4 linux/amd64


$ mysql --version
mysql  Ver 8.0.30 for Linux on x86_64 (MySQL Community Server - GPL)
```

### Operating System and Environment details

```sh
macOS Ventura Version 13.3.1

$ uname -m
arm64

$ uname -sr
Darwin 22.4.0
```

### Log Fragments

The following log fragment keeps repeating over and over in the `mysqld` container inside my `vttablet` pods:

```sh
I0515 18:32:52.217307       1 mysqld.go:425] Mysqld.Start(1684175472) stderr: 2023-05-15T18:32:52.204950Z mysqld_safe Number of processes running now: 0
I0515 18:32:52.217473       1 mysqld.go:431] Mysqld.Start(1684175472) stdout: 2023-05-15T18:32:52.204950Z mysqld_safe Number of processes running now: 0
I0515 18:32:52.486903       1 mysqld.go:425] Mysqld.Start(1684175472) stderr: 2023-05-15T18:32:52.472967Z mysqld_safe mysqld restarted
I0515 18:32:52.488177       1 mysqld.go:431] Mysqld.Start(1684175472) stdout: 2023-05-15T18:32:52.472967Z mysqld_safe mysqld restarted
I0515 18:32:54.596479       1 mysqld.go:425] Mysqld.Start(1684175472) stderr: 2023-05-15T18:32:54.501733Z 0 [Warning] [MY-011068] [Server] The syntax 'skip_slave_start' is deprecated and will be removed in a future release. Please use skip_replica_start instead.
I0515 18:32:54.597061       1 mysqld.go:425] Mysqld.Start(1684175472) stderr: 2023-05-15T18:32:54.523164Z 0 [Warning] [MY-010101] [Server] Insecure configuration for --secure-file-priv: Location is accessible to all OS users. Consider choosing a different directory.
I0515 18:32:54.597522       1 mysqld.go:425] Mysqld.Start(1684175472) stderr: 2023-05-15T18:32:54.526369Z 0 [Warning] [MY-010918] [Server] 'default_authentication_plugin' is deprecated and will be removed in a future release. Please use authentication_policy instead.
I0515 18:32:54.597868       1 mysqld.go:425] Mysqld.Start(1684175472) stderr: 2023-05-15T18:32:54.526731Z 0 [System] [MY-010116] [Server] /usr/sbin/mysqld (mysqld 8.0.30) starting as process 2086
I0515 18:32:54.767769       1 mysqld.go:425] Mysqld.Start(1684175472) stderr: 2023-05-15T18:32:54.763746Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
I0515 18:32:54.850400       1 mysqld.go:425] Mysqld.Start(1684175472) stderr: 2023-05-15T18:32:54.849762Z 1 [ERROR] [MY-012585] [InnoDB] Linux Native AIO interface is not supported on this platform. Please check your OS documentation and install appropriate binary of InnoDB.
I0515 18:32:54.851108       1 mysqld.go:425] Mysqld.Start(1684175472) stderr: 2023-05-15T18:32:54.850512Z 1 [Warning] [MY-012654] [InnoDB] Linux Native AIO disabled.
I0515 18:32:56.551551       1 mysqld.go:425] Mysqld.Start(1684175472) stderr: 2023-05-15T18:32:56.549191Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
I0515 18:32:57.642045       1 mysqld.go:425] Mysqld.Start(1684175472) stderr: 2023-05-15T18:32:57.641487Z 0 [Warning] [MY-010918] [Repl] 'rpl_semi_sync_master' is deprecated and will be removed in a future release. Please use rpl_semi_sync_source instead.
I0515 18:32:57.659752       1 mysqld.go:425] Mysqld.Start(1684175472) stderr: 2023-05-15T18:32:57.659533Z 0 [Warning] [MY-010918] [Repl] 'rpl_semi_sync_slave' is deprecated and will be removed in a future release. Please use rpl_semi_sync_replica instead.
I0515 18:32:57.826211       1 mysqld.go:425] Mysqld.Start(1684175472) stderr: 2023-05-15T18:32:57.825866Z 0 [System] [MY-010229] [Server] Starting XA crash recovery...
I0515 18:32:57.881003       1 mysqld.go:425] Mysqld.Start(1684175472) stderr: 2023-05-15T18:32:57.876664Z 0 [System] [MY-010232] [Server] XA crash recovery finished.
I0515 18:32:58.359945       1 mysqld.go:425] Mysqld.Start(1684175472) stderr: 2023-05-15T18:32:58.345770Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
I0515 18:32:58.360583       1 mysqld.go:425] Mysqld.Start(1684175472) stderr: 2023-05-15T18:32:58.348898Z 0 [System] [MY-013602] [Server] Channel mysql_main configured to support TLS. Encrypted connections are now supported for this channel.
I0515 18:32:58.391738       1 mysqld.go:425] Mysqld.Start(1684175472) stderr: 2023-05-15T18:32:58.372063Z 0 [Warning] [MY-011810] [Server] Insecure configuration for --pid-file: Location '/vt/vtdataroot' in the path is accessible to all OS users. Consider choosing a different directory.
I0515 18:32:58.894032       1 mysqld.go:425] Mysqld.Start(1684175472) stderr: 2023-05-15T18:32:58.872390Z 0 [System] [MY-010931] [Server] /usr/sbin/mysqld: ready for connections. Version: '8.0.30'  socket: '/vt/socket/mysql.sock'  port: 3306  MySQL Community Server - GPL.
I0515 18:33:06.155103       1 mysqld.go:425] Mysqld.Start(1684175472) stderr: Killed
```

**Comment by rohit-nayak-ps** [comment]
@vinimdocarmo, thanks for the detailed report and investigation. Please feel free to create a PR for this.

**Comment by vinimdocarmo** [comment]
@rohit-nayak-ps both PRs are mentioned in above.

**Comment by GuptaManan100** [comment]
We had a discussion about this issue, and we think the best way to fix this is to change the value for `innodb_buffer_pool_size` in MySQL. We have the field `configOverrides` in the mysql configuration for the operator.

**Comment by vinimdocarmo** [comment]
@GuptaManan100, did people in the discussion come up with a reasonable value for it? If not, I can try and experiment with some values.

But just to clarify: the idea is to have `mysqld` still using 512Mi and `minikube` with 8GB, but trying to decrease the amount of memory `mysqld` needs, by tweaking `innodb_buffer_pool_size` in this case, correct?

**Comment by GuptaManan100** [comment]
@vinimdocarmo You're understanding is correct! As far as the proposed value of `innodb_buffer_pool_size` is concerned, @mattlord would be the best person to advise on that. 
We can try experimenting with different values and see what works.

**Comment by GuptaManan100** [comment]
According to @mattlord `The default is 128MB, we could go all the way to 32MB`.

**Comment by drivebyer** [comment]
If you use MySQL 8.0, you can enable innodb-dedicated-server which configure innodb_buffer_pool_size value automaticlly.

**Comment by vinimdocarmo** [comment]
> According to @mattlord The default is 128MB, we could go all the way to 32MB.

@GuptaManan100 I tried 32MB. It _worked_ but not for the whole guide. As soon as I got to the  `kubectl -- apply -f 201_customer_tablets.yaml` command - where more tablets are created - some `mysqld` containers started failing for the same reason stated in the issue description: `Killed`. I even went down to 10MB, but I got to the same end. I checked minikube memory consumption and it was already floating right bellow the limit of 8GB.

I'm starting to think that tweaking `innodb_buffer_pool_size` only won't do the trick.

> If you use MySQL 8.0, you can enable innodb-dedicated-server which configure innodb_buffer_pool_size value automaticlly.

@drivebyer, thanks for the suggestion, but that wouldn't really help us. When less than 1GB of memory is available (which is our case), it will assign 128MB to `innodb_buffer_pool_size`, which is already the default.
